Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -1047,47 +1057,55 @@ private void InitialiseInterfaces()
                             // populate interface address list
                             foreach (UnicastIPAddressInformation info in ipProperties.UnicastAddresses)
                             {
-                                if (IsIP4Enabled && info.Address.AddressFamily == AddressFamily.InterNetwork)
+                                if (info.Address.AddressFamily == AddressFamily.InterNetwork)
                                 {
-                                    IPNetAddress nw = new IPNetAddress(info.Address, IPObject.MaskToCidr(info.IPv4Mask))
-                                    {
-                                        // Keep the number of gateways on this interface, along with its index.
-                                        Tag = ipProperties.GetIPv4Properties().Index
-                                    };
-
-                                    int tag = nw.Tag;
-                                    if (ipProperties.GatewayAddresses.Count > 0 && !nw.IsLoopback())
+                                    SystemIP4Enabled = true;
```

Comment:
This is assigned multiple times. Can it not be avoided?

Reply:
No. Only alternative is to assign true to a local variable then assign it back to the property.

SystemIP4Enabled is when the system has IP4 addresses (SystemIP6Enabled likewise). The only way I can safely calculate this is if the one of the network cards has an IP4 address. Hence it being set here.

Reply:
Isn't the IPv4 mapped to IPv6 only necessary for the _individual_ interfaces and not as a general thing? If Kestrel is listening on all interfaces, the proxy IP shouldn't  be mapped to IPv6 on an IPv4-only interface?

Reply:
There are two "levels" of IP in JF. User defined and system wide. SystemIP4Enabled/6Enabled - is the marker for the latter,

Kestral's IP address, although set by JF - goes to IP4 mapped if dual-stack is enabled. 

ListenOnIP will listen on both IP6/IP4 by default. This is one of the reasons we specify the interface addresses.

https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.kestrel.core.kestrelserveroptions.listenanyip?view=aspnetcore-5.0

Reply:
Do you by "system wide" mean that when IPv6 is enabled, then any IPv4-only interface will use mapped IPs?

What happens if JF is listening on two NICs, one is IPv4+IPv6 and the other is _only_ IPv4, and KnownProxies only uses mapped addresses?

Reply:
Ok just gone and tested.

If IP6Enabled is false - Kestrel never used IP6 addresses. The address list for listening is only IP4 addresses. (ListenOnIP is used)

If SystemIP6Enabled is false (no IP6 addresses on device) - the Kestrel cannot have any passed to it, so it's still in single-mode.

So in order for dual mode to come into effect - **any** interface needs an IP6 address, IP6 needs to be enabled (otherwise it's filtered).

Theoretically, it would be possible to enable IP6 on the device, enable it within JF, then only bind to IP4 addresses.
This would potentially cause an issue if either of these proxy settings were used.
I would argue that in this instance, it would be a config mismatch issue, and not a coding issue.

Hope that makes sense.